cms/djangoapps/modulestore_migrator/rest_api/v1/serializers.py

```diff
@@ -173,3 +175,63 @@ def get_fields(self):
         fields = super().get_fields()
         fields.pop('name', None)
         return fields
+
+
+class LibraryMigrationCourseSourceSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the source course of a library migration.
+    """
+    display_name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ModulestoreSource
+        fields = ['key', 'display_name']
+
+    def get_display_name(self, obj):
+        """
+        Return the display name of the source course
+        """
+        return self.context["course_names"].get(str(obj.key), None)
+
+
+class LibraryMigrationCollectionSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the target collection of a library migration.
+    """
+    class Meta:
+        model = Collection
+        fields = ["key", "title"]
+
+
+class LibraryMigrationCourseSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the course or legacylibrary migrations to V2 library.
+    """
+    source = LibraryMigrationCourseSourceSerializer()  # type: ignore[assignment]
+    target_collection = LibraryMigrationCollectionSerializer(required=False)
+    state = serializers.SerializerMethodField()
+    progress = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ModulestoreMigration
+        fields = [
+            'source',
+            'target_collection',
+            'state',
+            'progress',
+        ]
+
+    def get_state(self, obj: ModulestoreMigration):
+        """
+        Return the state of the migration.
+        """
+        if obj.is_failed:
+            return UserTaskStatus.FAILED
+
+        return obj.task_status.state
```

Comment:
I think we need to return `inProgress` in any case other than Success and Failed; to match [this type in the MFE](https://github.com/openedx/frontend-app-authoring/pull/2580/files#diff-263afcd9ce7c5d72ba9d15629098ce4968429582496f872a433bd9ed4bf2f4f8R801)